x-pack/plugins/lens/public/editor_frame_plugin/editor_frame/suggestion_panel.tsx

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import { Action } from './state_management';
+import { Datasource, Visualization, DatasourcePublicAPI } from '../../types';
+import { getSuggestions, toSwitchAction } from './suggestion_helpers';
+
+export interface SuggestionPanelProps {
+  activeDatasource: Datasource;
+  datasourceState: unknown;
+  activeVisualizationId: string | null;
+  visualizationMap: Record<string, Visualization>;
+  visualizationState: unknown;
+  datasourcePublicAPI: DatasourcePublicAPI;
+  dispatch: (action: Action) => void;
+}
+
+export function SuggestionPanel({
+  activeDatasource,
+  datasourceState,
+  activeVisualizationId,
+  visualizationMap,
+  visualizationState,
+  datasourcePublicAPI,
+  dispatch,
+}: SuggestionPanelProps) {
+  const datasourceSuggestions = activeDatasource.getDatasourceSuggestionsFromCurrentState(
+    datasourceState
+  );
+
+  const suggestions = getSuggestions(
+    datasourceSuggestions,
+    visualizationMap,
+    activeVisualizationId,
+    visualizationState,
+    datasourcePublicAPI
+  );
+
+  return (
+    <>
+      <h2>
+        <FormattedMessage
+          id="xpack.lens.editorFrame.suggestionPanelTitle"
+          defaultMessage="Suggestions"
+        />
+      </h2>
+      {suggestions.map((suggestion, index) => {
+        return (
+          <button
+            key={index}
```

Comment:
Index isn't a good key, and can lead to [subtle bugs](https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318). Is there a natural key we could use here? If not, it may be best to not use a key at all.

Reply:
There is no natural id (at least at the moment). But in the next step we will generate expressions for the suggestions, and then we can use the expression as an id (which is also nice because it prevents unnecessary re-renders if a suggestion just switches places with another one). I would take care of that in a separate PR.